cli/templatevariables.go

```diff
@@ -10,7 +10,21 @@ import (
 	"github.com/coder/coder/v2/codersdk"
 )
 
-func loadVariableValuesFromFile(variablesFile string) ([]codersdk.VariableValue, error) {
+func ParseUserVariableValues(variablesFile string, commandLineVariables []string) ([]codersdk.VariableValue, error) {
+	fromFile, err := parseVariableValuesFromFile(variablesFile)
+	if err != nil {
+		return nil, err
+	}
+
+	fromCommandLine, err := parseVariableValuesFromCommandLine(commandLineVariables)
+	if err != nil {
+		return nil, err
+	}
+
+	return combineVariableValues(fromFile, fromCommandLine), nil
```

Comment:
This raises an interesting question - what should the order of precedence be between values from files and values specified directly by CLI?
My hunch is that CLI variables should override those specified in files, and looking at `combineVariableValues` below that seems to be what would happen here.

Reply:
> My hunch is that CLI variables should override those specified in files, and looking at combineVariableValues below that seems to be what would happen here.

Yeah, that is my idea too. I described the order of precedence [here](https://github.com/coder/coder/issues/8501#issuecomment-1855663509).